lib/src/message/message_item_bloc.dart

```diff
@@ -228,6 +234,12 @@ class MessageItemBloc extends Bloc<MessageItemEvent, MessageItemState> {
           unregisterListeners();
         }
       }
+    }else if (event.hasType(Event.msgFailed)){
+      int eventMessageState = ChatMsg.messageStateFailed;
+      Context context = Context();
+      String messageInfo = await context.getMessageInfo(_messageId);
+      var messageStateData = (state as MessageItemStateSuccess).messageStateData.copyWith(state: eventMessageState, messageInfo: messageInfo);
+      add(MessageUpdated(messageStateData: messageStateData));
```

Comment:
As this "failed state" is final we should probably call `unregisterListeners();` here.

Reply:
Right.